src/main/java/net/earthcomputer/clientcommands/c2c/CCNetworkHandler.java

```diff
@@ -0,0 +1,76 @@
+package net.earthcomputer.clientcommands.c2c;
+
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import com.mojang.datafixers.kinds.IdF;
+import net.earthcomputer.clientcommands.c2c.packets.MessageC2CPacket;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.PlayerListEntry;
+import net.minecraft.network.encryption.PlayerPublicKey;
+import net.minecraft.text.MutableText;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+
+import java.security.PublicKey;
+
+public class CCNetworkHandler implements CCPacketListener {
+
+    private static final SimpleCommandExceptionType MESSAGE_TOO_LONG_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("ccpacket.messageTooLong"));
+    private static final SimpleCommandExceptionType PUBLIC_KEY_NOT_FOUND_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("ccpacket.publicKeyNotFound"));
+    private static final SimpleCommandExceptionType ENCRYPTION_FAILED_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("ccpacket.encryptionFailed"));
+
+    public static boolean justSent = false;
+
+    private static CCNetworkHandler instance;
+
+    private CCNetworkHandler() {
+    }
+
+    public static CCNetworkHandler getInstance() {
+        if (instance == null) {
+            instance = new CCNetworkHandler();
```

Comment:
The instance field probably doesn't need to be lazy (and can therefore be final)